heroic-component/src/main/java/com/spotify/heroic/cluster/ClusterShard.java

```diff
@@ -47,16 +50,16 @@
     private final AsyncFramework async;
 
     private final Map<String, String> shard;
-    private final List<ClusterNode.Group> groups;
     private final QueryReporter reporter;
+    private final ClusterManager cluster;
+
+    private final List<UUID> nodesTried = new ArrayList<>();
```

Comment:
This should be inside `apply(..)`, otherwise its state would be shared across requests over one specific ClusterShard.